pyvista/plotting/plotting.py

```diff
@@ -4832,3 +4849,30 @@ def add_title(self, title, font_size=18, color=None, font=None,
 # Tracks created plotters.  At the end of the file as we need to
 # define ``BasePlotter`` before including it in the type definition.
 _ALL_PLOTTERS: Dict[str, BasePlotter] = {}
+
+
+def _kill_display(disp_id):  # pragma: no cover
+    """Forcibly close the display on Linux.
+
+    See:
+    https://gitlab.kitware.com/vtk/vtk/-/issues/17917#note_783584
+
+    And more details into why...
+    https://stackoverflow.com/questions/64811503
+
+    # this is to be used experimentally and is known to cause issues
+    # on `pyvistaqt`
+
+    """
+    if platform.system() != 'Linux':
+        raise OSError('This method only works on Linux')
+
+    if disp_id:
+        cdisp_id = ctypes.c_size_t(int(disp_id[1:].split('_')[0], 16))
+
+        # check display can be closed and then close it in the background
+        # if not X11.XEventsQueued(cdisp_id, 0):
+
+        # this is unsafe as events might be queued, but sometimes the
+        # window fails to close without this
+        Thread(target=X11.XCloseDisplay, args=(cdisp_id, )).start()
```

Comment:
Why not set `X11.XCloseDisplay.argtypes` as suggested in https://stackoverflow.com/questions/64811503/python-memory-addresses-variation-across-python-3-5-3-9-on-linux#comment114613236_64814361, to avoid casting to a `c_size_t` each time?